index.bs

```diff
@@ -822,9 +836,22 @@ The origin of a {{XRBoundedReferenceSpace}} MUST be positioned at the floor, suc
 
 Note: Other XR platforms sometimes refer to the type of tracking offered by a {{bounded}} reference space as "room scale" tracking. An {{XRBoundedReferenceSpace}} is not intended to describe multi-room spaces, areas with uneven floor levels, or very large open areas. Content that needs to handle those scenarios should use an {{unbounded}} reference space.
 
+The <dfn constructor for="XRBoundedReferenceSpace">XRBoundedReferenceSpace(|base|, |originOffset|)</dfn> constructor MUST perform the following steps when invoked:
+
+  1. Let |refSpace| be a new {{XRBoundedReferenceSpace}}.
+  1. If |base|'s {{XRReferenceSpace/base}} is <code>null</code>, set |refSpace|'s {{XRReferenceSpace/base}} to |base|.
+  1. Else set |refSpace|'s {{XRReferenceSpace/base}} to |base|'s {{XRReferenceSpace/base}}.
+  1. Set |refSpace|'s {{XRReferenceSpace/originOffset}} to |originOffset| multiplied by |base|'s {{XRReferenceSpace/originOffset}}.
```

Comment:
Is there an existing definition what it means to multiply XRRigidTransforms? This needs to be unambiguous. I think it's backwards as written.

Let's say we set up a base space and two chained offsets like this:

```js
let baseSpaceA = ...; starting reference space A
let offsetSpaceA = new XRBoundedReferenceSpace(baseSpaceA, originOffsetA);
let offsetSpaceB = new XRBoundedReferenceSpace(offsetSpaceA, originOffsetB);
```

Each originOffset's matrix is a transform from offset space to base space. Space B's base space corresponds to space A's offset space.

You can use the matrices to transform coordinates in `offsetSpaceB` to coordinates in `baseSpaceA`:

```
pointInBaseA = baseAFromOffsetA * offsetAFromOffsetB * pointInOffsetB;
             = offsetSpaceA.originOffset.matrix * offsetSpaceB.originOffset.matrix * pointInOffsetB
```


and therefore, space B's originOffset.matrix is space A's originOffset.matrix times space B's originOffset.matrix, or for spec purposes:

```
1. Set |refSpace|'s {{XRReferenceSpace/originOffset}} to |base|'s {{XRReferenceSpace/originOffset}} multiplied by |originOffset|.
```

Reply:
➕ 1 to having this defined as an "algorithm" in the spec so we can easily reference it in other spec "algorithms".  @klausw , do you want to post a PR with this clarification text and Brandon and I can review it?  I think getting that nailed down first will go a long way towards our goal of algorithmically defining the behavior of `getPose()` and `getViewerPose()`.

Reply:
@NellWaliczek That's what I was trying to do in my existing PR, see especially https://github.com/immersive-web/webxr/pull/569/files#diff-ec9cfa5f3f35ec1f84feb2e59686c34dR955 which tries to be quite specific about this.

Can we discuss that in that PR? I'm happy to extend this further, i.e. adding more info about chaining transforms, but I'd like to get consensus on the basics first.

Reply:
Sorry, I think there's a misunderstanding here between Nell and Klaus. I believe the algorithm which Nell is talking about is one that describes how two XRRigidTransforms are multiplied. (At least that's how it reads given the context.) The line in Klaus' PR that was linked details (very well) how a matrix is derived from the rigid transform.

I do think that having an in-spec algorithm that describes how transforms are multiplied would be extremely useful. That's something that Nell or I could sketch out if needed, but either way I'd want to have Klaus and/or Manish review it to ensure we're all on the same page. Then that should serve as a fundamental building block though which we can help describe a great many of the rest of these concepts.

Reply:
@toji The text I have now in #569 specifies the content of a XRRigidTransform matrix and how it transforms numeric coordinates, and this should constrain implementations to one specific interpretation of how the transforms work. As a result, there should only be one correct way to chain transforms.

I don't think it makes sense to have an independent PR or discussion about chaining transforms unless we can get agreement on these basic properties of transforms. There's no separate design freedom for the underlying matrix math of XRRigidTransform multiplication.  I agree it would be helpful to add an explanation to the text, but I think this would be appropriate as a followup since it's derived from the basic properties and definitions of the transforms and matrices.

I'm fairly confident that the chaining I described in [my comment above](https://github.com/immersive-web/webxr/pull/589#discussion_r275979306) matches #569, assuming usual column-major GL matrices applied to column vectors, but I'd appreciate an independent sanity check.

Would it help to schedule some time with interested parties to talk through this? I feel as if we're still a bit stuck on this issue even after discussing it in two of the weekly conf calls, and I'm unsure if there's actual disagreement or not.